fe/fe-core/src/main/java/com/starrocks/sql/optimizer/ChildOutputPropertyGuarantor.java

```diff
@@ -390,6 +453,23 @@ public Void visitPhysicalJoin(PhysicalJoinOperator node, ExpressionContext conte
                             rightChild, rightChildOutputProperty);
                 }
                 return visitOperator(node, context);
+            } else if (leftDistributionDesc.isIcebergLocal() && rightDistributionDesc.isIcebergLocal()) {
```

Comment:
think about local olap join with local iceberg. Maybe refractor this or it will be too many branches.

Reply:
done